scripts/gha/test_simulator.py

```diff
@@ -12,57 +12,81 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-r"""Tool for mobile testapps to Test on local simulators.
-
-Requires simulators installed. iOS simulator can be installed via tool:
-  https://github.com/xcpretty/xcode-install#simulators
-
+r"""Tool for mobile testapps to Test on local simulators/emulaotrs.
 
 Usage:
 
   python test_simulator.py --testapp_dir ~/testapps 
 
 This will recursively search ~/testapps for apps,
-test on local simulator, and validate their results. The validation is specific to
-the structure of the Firebase Unity and C++ testapps.
+test on local simulators/emulaotrs, and validate their results. The validation is 
+specific to the structure of the Firebase Unity and C++ testapps.
 
+----iOS only----
+Requires simulators installed. iOS simulator can be installed via tool:
+  https://github.com/xcpretty/xcode-install#simulators
 
-If you wish to specify a particular device to test on, you will need the model
+If you wish to specify a particular iOS device to test on, you will need the model
 id and version (OS version for iOS). These change over time. You can listing all 
 available simulators (supported models and versions) with the following commands:
 
   xcrun simctl list
 
-Note: you need to combine Name and Version with "-". Examples:
-
+Note: you need to combine Name and Version with ";". Examples:
 iPhone 11, OS 14.4:
-  --ios_device "iPhone 11-14.4"
+  --ios_device "iPhone 11;14.4"
+
+----Android only----
+Java 8 is required
+Environment Variables (on MacOS)
+    JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-8-latest/Contents/Home
+    ANDROID_HOME=/Users/user_name/Library/Android/sdk
+Environment Variables (on Linux)
+    JAVA_HOME=/usr/local/buildtools/java/jdk8/
+    ANDROID_HOME=~/Android/Sdk
+
+If you wish to specify a particular Android device to test on, you will need 
+the sdk id and build tool version. These change over time. You can listing all 
+available tools with the following commands:
+
+  $ANDROID_HOME/tools/bin/sdkmanager --list
+
+Note: you need to combine them with ";". Examples:
+
+sdk id "system-images;android-29;google_apis", build tool "29.0.3":
+  --android_device "system-images;android-29;google_apis;x86;29.0.3"
```

Comment:
This script has special return error codes. Document various error codes and a brief description.

Reply:
I reused same error code in different cases. 
Is `logging.error` enough to provide a brief description? Any ideas?

Reply:
We could do something like the module level docstring here,
https://github.com/firebase/firebase-cpp-sdk/blob/003bf4cfddb8dbdc5a19612caccd36c8d5d7e7b8/scripts/format_code.py